### PR TITLE
WIP Entry scripts

### LINKED ISSUE #27: Entry script for Meson nodes
Currently, there are a lot of configuration files with unique values. These values are currently maintained manually in both the config files and docker-compose. The parameters are made up of:
- ip address
- public ED25519 keys, which are derived from private key files.

To keep a single source of truth for these values I think a solution would be to create an entry script that runs `sed` on the config files. the new values are determined by environment variables in the docker-compose.

### PR DESCRIPTION
**Pull request by sebohe**
This PR closes #27 #47.

This PR makes the containers more modular by generating the config files through ENVIRONMENT variables  the config files are generate during the container startup or you can also reference. This makes the containers less dependent on the docker compose hard coded values to coordinate between the nodes and instead just needs environments variables to be set when running the container. These environments variables can later be changed to command line flags to avoid the awkward syntax of using environment variables.

Changes:
- Config files can be generate from the command line flags or from mounting an existing config file into the right path. Currently the default path is `
- New `dynamic-compose.yml` that makes use of the environment variables. It also doesn't use geth nodes
- Added one additional container image: `hashcloak/authority`.
- The log files go to `/dev/stdout`
- The `hashcloak/provider` contains both the config files for being a provider or a mix node.

TODO:
- [ ] Provide instructions on how to use the containers in standalone manner.
- [x] Need to improve the logic on how required environment variables are checked.
- [ ] Make sure that the original `docker-compose` works well for local dev work without the need of the environment variables or the too fancy docker scripts.
- [ ] ~~Make sure all log files go to `/dev/stdout`~~ This is going to be harder to implement because the plugins use a `log_dir` instead of a `log_file`.

### CODE REVIEW

File: ops/dynamic-testnet.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    image: ${MESON_IMAGE}
+    environment:
+      - AUTHORITY_KEY=o4w1Nyj/nKNwho5SWfAIfh7SMU8FRx52nMHGgYsMHqQ=
+      - AUTHORITY_ADDRESS=172.28.1.2:29483
+      - IDENTIFIER=provider1
+      - PORT=59483
+      - IS_PROVIDER=true
+      - LOG_LEVEL=DEBUG
+      - DISABLE_LOGS=false
+      - RPC_URL=https://g.sebas.tech
```
Comment (Mikerah) [inline ops/dynamic-testnet.yml]: Add a note that this is your own hardcoded computer.

File: Makefile [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
 katzenAuth=$(dockerRepo)/katzenpost-auth:$(katzenAuthTag)
 
 gethVersion=v1.9.9
 gethImage=$(dockerRepo)/client-go:$(gethVersion)
 mesonServer=$(dockerRepo)/meson
-mesonClient=$(dockerRepo)/meson-client
+hashcloakAuth=$(dockerRepo)/authority
 
-messagePush="LOG: Image already exists in docker.io/$(repo). Not pushing: "
+messagePush="LOG: Image already exists in docker.io/$(repoRepo). Not pushing: "
```
Comment (Mikerah) [inline Makefile]: `repoRepo` isn't defined anywhere else in this makefile. What is it suppose to point to? Also, why name it `repoRepo`? Did the previous name `repo` conflict with a previous make variable?

Comment (sebohe) [reply Makefile]: I used the wrong variable. The correct one is `dockerRepo`

**Comment by sebohe** [comment]
Closing this issue and all other entry scripts issues because they currently don't support the modularity that we need at this moment.

**Pull request closed without merging**